build/mongosh/test.js.example

```diff
@@ -0,0 +1,31 @@
+const assert = require('assert');
+
+(function() {
+    "use strict";
+    const coll = db.issue1539;
+
+    coll.drop();
+    
+    docs = [
+      { _id: "double", v: 42.13 },
+      { _id: "double-whole", v: 42.0 },
+      { _id: "double-zero", v: 0.0 },
+      { _id: "double-max", v: Number.MAX_VALUE },
+      { _id: "double-smallest", v: Number.MIN_VALUE },
+      { _id: "doubleBig", v: Number(2 << 60) },
+      { _id: "double-null", v: null }
+    ]
+
+    coll.insertMany(docs);
+
+    let noError = ""; // bad error
+    let expectedError = "Failed to parse $size. Expected a non-negative number in: $size: -1"; // good error
+
+    x = { v: { $size: -1 } };
+    try {
+        coll.findOne(x);
+    } catch(error) {
+        assert.equal(error.message, expectedError)
```

Comment:
If we are expecting error and if `coll.findOne(x)` doesn't return error, we won't notice it. Is there a way to check that error is really caught?

Reply:
Nice catch. We can use a `finally` block after the `catch` to indicate that no exception was caught and then print something. We should also compare error codes instead of messages as discussed.

Reply:
Let me add corrections here. `finally` will not work for this. If the method returns an exception then one will be caught, we can check if the error is `null` and then print out a statement.

Reply:
Something like this @chilagrow?

```javascript
let expectedErrorCode = 2;
let codeName = 'BadValue'; // maybe we can use this somehow

error = null;
x = { v: { $size: -1 } };
y = {}; // failing test
try {
    coll.findOne(y);
} catch(e) {
    error = e;
    assert.equal(error.code, expectedErrorCode)
}
if (error == null) {
    console.log("test failed")
}
```

Reply:
Looks good :+1: